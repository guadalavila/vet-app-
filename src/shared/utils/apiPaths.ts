export const API_PATHS = {
    SIGN_UP: 'auth/signup',
    LOGIN: 'auth/login',
    GET_ME: 'auth/get-me',
    DASHBOARD: 'dashboard',
    PETS: 'pets',
    ONE_PET: 'pets/one/',
    PETS_BY_VET: 'pets/by-vet/',
    PETS_BY_CLIENT: 'pets/by-client/',
    PET: 'pet',
    CLIENTS: 'clients',
    CLIENTS_BY_VET: '/clients/by-vet/',
    CLIENT: 'clients/',
    SEARCH_CLIENTS: 'clients/search/',
    SEARCH_ONE_CLIENT: 'clients/search/one/',
    VISITS: 'visits',
    VISIT_BY_PET: 'visits/by-pet/',
    LAST_VISIT: 'visits/last/',
    LAST_VISIT_BY_VET: 'visits/by-vet/',
    CONDITIONS: 'conditions',
    PATHOLOGIES: 'pathologies',
    SEARCH: '/search/',
    VACCINE: 'vaccine',
    VACCINE_BY_PET: 'vaccine/by-pet/',
    REPORT_BY_VET: 'report/by-vet/',
    SURGERY: 'surgery',
    SURGERIES_BY_PET: 'surgery/by-pet/',
    VETERINARY: 'vets',
    USERS: 'users',
};
