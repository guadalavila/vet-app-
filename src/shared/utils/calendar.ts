type CONFIG_MONTHS_DAYS_PROPS = {
    monthNames: string[],
    monthNamesShort: string[],
    dayNames: string[],
    dayNamesShort: string[],
    today: string,
};

const monthNames: string[] = [
    'Enero',
    'Febrero',
    'Marzo',
    'Abril',
    'Mayo',
    'Junio',
    'Julio',
    'Agosto',
    'Septiembre',
    'Octubre',
    'Noviembre',
    'Diciembre',
];

const monthNamesShort: string[] = ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun', 'Jul', 'Ago', 'Sep', 'Oct', 'Nov', 'Dic'];

const dayNames: string[] = ['Domingo', 'Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes', 'Sábado'];

const dayNamesShort: string[] = ['D', 'L', 'M', 'M', 'J', 'V', 'S'];

const today: string = 'Hoy';

export const LOCALE_CONFIG = 'es';

export const CONFIG_MONTHS_DAYS: CONFIG_MONTHS_DAYS_PROPS = {
    monthNames,
    monthNamesShort,
    dayNames,
    dayNamesShort,
    today,
};
