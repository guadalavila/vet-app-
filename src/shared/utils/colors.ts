import { Theme } from 'react-native-calendars/src/types';
import { typography } from './typography';

const common = {
    // primary: '#FD9340',
    primary: '#917FB3',

    // secondary: '#2A2F4F',
    // tertiary: '#E5BEEC',
    quaternary: '#FDE2F3',
    white: '#FFFFFF',
    black: '#2F2F2F',
    error: '#E74C3C',
    grey: '#EEEEEE',
    greySecondary: '#f5f5f5',
    greyDark: '#A3A3A3',
    whatsapp: '#25d366',
    success: '#4E9F3D',
    teal: '#73B4B0',
};

const light = {
    ...common,
    background: '#FFFFFF',
    text: '#11100F',
    secondary: '#2A2F4F',
    tertiary: '#E5BEEC',
    container: '#f5f4f6',
    separator: '#F1f1f1',
    border: '#2A2F4F',
    input: '#f2f1f6',
    textInput: '#515151',
    bottomSheet: '#F1f1f1',
    card: '#D6D6D6',
    backgroundInput: '#f6f8fe',
};

const dark = {
    ...common,
    background: '#131313',
    text: '#F1F1F1',
    tertiary: '#2A2F4F',
    secondary: '#E5BEEC',
    container: '#222125',
    separator: '#f0f0f0',
    border: '#E5BEEC',
    input: '#4C4C4C',
    textInput: '#DEDEDE',
    bottomSheet: '#71717C',
    card: '#D6D6D6',
    backgroundInput: '#414141',
};

export const colors = { light, dark };

export const THEME_CUSTOM_CALENDAR: Theme = {
    textSectionTitleColor: colors.light.primary,
    textDayHeaderFontSize: typography.size.M,
    calendarBackground: colors.light.grey,
    todayTextColor: colors.light.black,
    arrowHeight: 50,
    textMonthFontSize: typography.size.S,
    textDayHeaderFontWeight: 'bold',
    arrowColor: colors.light.primary,
    dayTextColor: colors.light.black,
    monthTextColor: colors.light.primary,
    selectedDayTextColor: 'white',
    selectedDayBackgroundColor: colors.light.primary,
    textDisabledColor: colors.light.grey,

    // selectedDayBackgroundColor: '#333248',
};
