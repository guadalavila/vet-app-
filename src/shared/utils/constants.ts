import { Categorie } from '../../models/Categorie';

export const CATEGORIES: Categorie[] = [
    { id: 1, name: 'CLIENTES', page: 'ClientsScreen', icon: 'people-outline' },
    { id: 1, name: 'MASCOTAS', page: 'PetsScreen', icon: 'paw-outline' },
];
