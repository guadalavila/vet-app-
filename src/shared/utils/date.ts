export const getPreviousDay = (date = new Date()) => {
    const previous = new Date(date.getTime());
    previous.setDate(date.getDate() - 1);
    return previous;
};

function padTo2Digits(num: number) {
    return num.toString().padStart(2, '0');
}

export const formatDate = (date: Date) => {
    // date.getFullYear()
    return [padTo2Digits(date.getDate()), padTo2Digits(date.getMonth() + 1)].join('/');
};

export const isSelectedDate = (dateA: Date, dateB: Date) => {
    return formatDate(dateA) === formatDate(dateB);
};

export const formatDateCalendar = (date: Date) => {
    return [date.getFullYear(), padTo2Digits(date.getMonth() + 1), padTo2Digits(date.getDate())].join('-');
};

export const getFirstDayOfMonth = () => {
    const date = new Date();
    return new Date(date.getFullYear(), date.getMonth() - 1, 1);
};
