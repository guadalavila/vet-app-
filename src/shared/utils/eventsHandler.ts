class AppEventsHandler<EventType> {
    // eslint-disable-next-line prettier/prettier
    private subscribers: Record<keyof EventType, Set<(data: EventType[any]) => void>> = {} as any;

    listen<K extends keyof EventType>(type: K, cb: (data: EventType[K]) => void): () => void {
        if (!this.subscribers[type]) {
            this.subscribers[type] = new Set([]);
        }
        if (this.subscribers[type].has(cb)) {
            return () => {
                this.subscribers[type].delete(cb);
            };
        }
        this.subscribers[type].add(cb);
        return () => {
            this.subscribers[type].delete(cb);
        };
    }

    publish<K extends keyof EventType>(type: K, data: EventType[K]): void {
        if (!this.subscribers[type]) {
            return;
        }
        this.subscribers[type].forEach((cb) => {
            cb(data);
        });
    }
}
export default AppEventsHandler;
