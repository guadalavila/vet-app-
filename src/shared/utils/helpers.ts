const mapPetType = () => {
    const mapGender = new Map();
    mapGender.set('birds', 'Aves');
    mapGender.set('canine', 'Canino');
    mapGender.set('equine', 'Equino');
    mapGender.set('feline', 'Felino');
    mapGender.set('ferret', 'Huron');
    mapGender.set('insect', 'Insecto');
    mapGender.set('rodent', 'Roedor');
    mapGender.set('unknown', 'Desconocido');
    mapGender.set('', '');
    return mapGender;
};

export const getPetType = (type: string) => {
    return mapPetType().get(type);
};

const mapPetSize = () => {
    const mapSize = new Map();
    mapSize.set('small', 'Chico');
    mapSize.set('medium', 'Mediano');
    mapSize.set('big', 'Grande');
    mapSize.set('other', 'Otro');
    mapSize.set('', '');
    return mapSize;
};

export const getPetSize = (size: string) => {
    return mapPetSize().get(size);
};

const mapPetGender = () => {
    const mapGender = new Map();
    mapGender.set('male', 'Macho');
    mapGender.set('female', 'Hembra');
    mapGender.set('unknow', 'Desconocido');
    mapGender.set('', '');
    return mapGender;
};

export const getPetGender = (gender: string) => {
    return mapPetGender().get(gender);
};
