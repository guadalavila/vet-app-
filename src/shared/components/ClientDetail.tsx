import React from 'react';
import { Linking, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Card from './Card';
import CustomText from './CustomText';
import { Client } from '../../models/Client';
import Icon from 'react-native-vector-icons/Ionicons';
import { size } from '../utils/size';
import { typography } from '../utils/typography';
import { colors } from '../utils/colors';
import { GlobalStyles } from '../utils/styles';

interface IClientDetailProps {
    client: Client;
}

const ClientDetail = ({ client }: IClientDetailProps) => {
    const getCodeName = () => client.name.charAt(0).concat(client.lastName.charAt(0)).toUpperCase();
    return (
        <Card>
            <View style={styles.containerAvatar}>
                <Text style={styles.textAvatar}>{getCodeName()}</Text>
            </View>
            <CustomText style={[styles.name]}>
                {client.name} {client.lastName}
            </CustomText>
            <CustomText style={[styles.dni]}>DNI {client.dni}</CustomText>
            {client.email && client.email.length > 0 && (
                <CustomText style={styles.emailText}> {client.email}</CustomText>
            )}
            {client.address && client.address.length > 0 && (
                <CustomText style={styles.emailText}> {client.address}</CustomText>
            )}
            <TouchableOpacity
                style={[GlobalStyles.row, styles.containerPhone]}
                activeOpacity={0.7}
                onPress={() => Linking.openURL(`https://api.whatsapp.com/send?phone=${client.phone}}`)}>
                <Icon name='logo-whatsapp' size={24} color={colors.light.whatsapp} />
                <CustomText style={styles.phone}>{client.phone}</CustomText>
            </TouchableOpacity>
        </Card>
    );
};

export default ClientDetail;

const styles = StyleSheet.create({
    name: {
        color: colors.light.primary,
        fontSize: typography.size.L,
        textAlign: 'center',
        fontWeight: 'bold',
        marginVertical: size.XXL,
    },
    dni: {
        fontSize: typography.size.M,
        textAlign: 'center',
        marginVertical: size.L,
    },
    containerPhone: {
        marginVertical: size.XL,
        alignSelf: 'center',
    },
    phone: {
        fontWeight: 'bold',
        marginLeft: size.L,
        alignSelf: 'center',
        fontSize: typography.size.S,
        textDecorationLine: 'underline',
    },
    containerAvatar: {
        backgroundColor: colors.light.primary,
        borderRadius: 100,
        width: 100,
        height: 100,
        alignSelf: 'center',
        justifyContent: 'center',
        marginTop: size.L,
    },
    textAvatar: {
        fontSize: typography.size.XXXL,
        color: colors.light.white,
        alignSelf: 'center',
    },
    emailText: {
        marginHorizontal: size.L,
        marginBottom: size.XL,
        textAlign: 'center',
    },
});
