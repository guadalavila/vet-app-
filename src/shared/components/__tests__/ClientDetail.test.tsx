import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import ClientDetail from '../ClientDetail';
import { Client } from '~models/Client';
import { colors } from '~shared/utils/colors';
import { Linking } from 'react-native';

describe('ClientDetail', () => {
    test('should match snapshot', () => {
        const container = render(<ClientDetail client={mockClient} />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<ClientDetail client={mockClient} />);
        expect(screen.getByTestId('ClientCard')).toBeTruthy();
    });

    test('should render texts', () => {
        render(<ClientDetail client={mockClient} />);
        expect(screen.getByText('JP')).toBeTruthy();
        expect(screen.getByTestId('nameAndLastName').children).toEqual(['Juan', ' ', 'Pérez']);
        expect(screen.getByTestId('dni').children).toEqual(['DNI ', '12345678A']);
        expect(screen.getByTestId('email').children).toEqual(['juan@example.com']);
        expect(screen.getByTestId('address').children).toEqual(['Calle Falsa 123']);
    });

    test('should render phone and whatsapp icon', () => {
        render(<ClientDetail client={mockClient} />);
        expect(screen.getByTestId('phone-button')).toBeTruthy();
        expect(screen.getByTestId('icon')).toBeTruthy();
        expect(screen.getByTestId('icon').props.name).toEqual('logo-whatsapp');
        expect(screen.getByTestId('icon').props.size).toEqual(24);
        expect(screen.getByTestId('icon').props.color).toEqual(colors.light.whatsapp);
        expect(screen.getByTestId('phone').children).toEqual(['123-456-7890']);
    });

    test('should call a function when press phone', () => {
        render(<ClientDetail client={mockClient} />);
        const button = screen.getByTestId('phone-button');
        fireEvent.press(button);
        expect(Linking.openURL).toHaveBeenCalledWith(`https://api.whatsapp.com/send?phone=${mockClient.phone}`);
    });
});

const mockClient: Client = {
    _id: '123456789',
    vetId: 'vet123',
    dni: '12345678A',
    name: 'Juan',
    lastName: 'Pérez',
    email: 'juan@example.com',
    phone: '123-456-7890',
    address: 'Calle Falsa 123',
    comment: 'Preferred contact method: email',
    createdAt: '2023-08-26T10:00:00Z',
    updatedAt: '2023-08-26T14:30:00Z',
};
