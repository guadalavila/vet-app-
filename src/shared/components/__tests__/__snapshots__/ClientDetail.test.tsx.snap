// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientDetail should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 16,
        "marginHorizontal": 14,
        "paddingHorizontal": 8,
        "paddingVertical": 10,
      },
      Object {
        "elevation": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      Object {
        "backgroundColor": "#f5f4f6",
      },
    ]
  }
  testID="ClientCard"
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#A57FEE",
        "borderRadius": 100,
        "height": 100,
        "justifyContent": "center",
        "marginTop": 8,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontSize": 48,
        }
      }
    >
      JP
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#A57FEE",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 14,
            "textAlign": "center",
          },
        ],
      ]
    }
    testID="nameAndLastName"
  >
    Juan
     
    PÃ©rez
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Array [
          Object {
            "fontSize": 20,
            "marginVertical": 8,
            "textAlign": "center",
          },
        ],
      ]
    }
    testID="dni"
  >
    DNI 
    12345678A
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {
          "marginBottom": 10,
          "marginHorizontal": 8,
          "textAlign": "center",
        },
      ]
    }
    testID="email"
  >
    juan@example.com
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {
          "marginBottom": 10,
          "marginHorizontal": 8,
          "textAlign": "center",
        },
      ]
    }
    testID="address"
  >
    Calle Falsa 123
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "marginVertical": 10,
        "opacity": 1,
      }
    }
    testID="phone-button"
  >
    <MockedIoniconsIcons
      color="#25d366"
      name="logo-whatsapp"
      size={24}
      testID="icon"
    />
    <Text
      style={
        Array [
          Object {
            "color": "#11100F",
            "fontSize": 16,
          },
          Object {
            "alignSelf": "center",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 8,
            "textDecorationLine": "underline",
          },
        ]
      }
      testID="phone"
    >
      +5491149876542
    </Text>
  </View>
</View>
`;
